tools/guided-fuzzing/src/GuidedExecutionPolicy.cpp

```diff
@@ -0,0 +1,26 @@
+#include "include/GuidedExecutionPolicy.h"
+
+namespace caffeine {
+
+GuidedExecutionPolicy::GuidedExecutionPolicy(AssertionList& list,
+                                             std::shared_ptr<Solver> solver)
+    : requiredAssertions_{list}, solver_{solver} {
+  if (solver_->check(list) == UNSAT) {
+    CAFFEINE_ABORT("GuidedExecutionPolicy requires satisfiable premise");
+  }
+}
+
+bool GuidedExecutionPolicy::should_queue_path(const Context& ctx) {
+  AssertionList combined;
+  for (const auto& i : ctx.assertions) {
+    combined.insert(i);
+  }
```

Comment:
Just copying the list keeps the internal state better and is probably faster.
```suggestion
  AssertionList combined = ctx.assertions;
```